### ISSUE TITLE
Icon-only controls are not accessible

### ISSUE THREAD

**Issue by afercia** [WordPress/twentytwenty#93]
In the current version of the theme c277a895704cc00ec402261a7f30fdd409d03ffa the Search and Menu controls in the header use both visible text and icons:

<img width="1041" alt="Screenshot 2019-09-07 at 19 38 34" src="https://user-images.githubusercontent.com/1682452/64478613-d635b400-d1aa-11e9-8d60-9f8919f83cf5.png">

These come from [the Chaplin theme](https://wordpress.org/themes/chaplin/) and are perfectly usable by all users.

Instead, in the design shown in [the Twenty Twenty post on Make](https://make.wordpress.org/core/2019/09/06/introducing-twenty-twenty/) the header design is different and uses icon-only controls. This new design is going to be implemented in https://github.com/WordPress/twentytwenty/pull/71

<img width="962" alt="header order" src="https://user-images.githubusercontent.com/1682452/64478418-74744a80-d1a8-11e9-8a59-974a9694bc26.png">

However, icon-only controls are not accessible. [Icons don't have a universal meaning](https://www.nngroup.com/articles/icon-usability/). Not all users can operate on UI controls that don't expose their accessible name visually.

The same issue was discussed at length in the Gutenberg project. "icon-only" controls are problematic for many reasons. They're an accessibility anti-pattern as pointed out in the first [Gutenberg accessibility report](https://make.wordpress.org/accessibility/2018/10/29/report-on-the-accessibility-status-of-gutenberg/) published by the accessibility team on October 2018.

For example, speech recognition software users need to know the accessible name (the label of control's text) to be able to voice a command the software can understand, as well explained in https://github.com/WordPress/gutenberg/issues/470#issuecomment-435611115

A few previous related comments on the Gutenberg repository:
- https://github.com/WordPress/gutenberg/issues/5358#issuecomment-371070915
- https://github.com/WordPress/gutenberg/issues/7221#issuecomment-398748271

Similarly, the recent WPCampus Gutenberg accessibility audit by Tenon pointed out the same issue in various part of the WordPress interface. See for example:

Visible labels are not available on editing buttons
https://github.com/WordPress/gutenberg/issues/15352
then moved to Trac https://core.trac.wordpress.org/ticket/47116

Icons alone are visible for controls unless they are hovered or focussed
https://github.com/WordPress/gutenberg/issues/15311

In both cases, the provided remediation guideline is to add visible text labels to the buttons or a setting so that users can specify whether to show icons-only, labels-only, or icons and labels together. For what concerns Gutenberg, the second option is currently explored in a 
 pending pull request, see https://github.com/WordPress/gutenberg/issues/10524 and https://github.com/WordPress/gutenberg/pull/15830. I do realise user settings on a theme would be uncommon. The alternative is to add visible text.

**Comment by andersnoren** [comment]
Hi @afercia,

Thank you for posting this issue. I was hoping to get back to you earlier, but things got in they way. We've talked a bit about this now, and decided that we're going to include visible text for the icons in the header (including the close search button).

On desktop, when there's a menu set to the "Shortcuts" menu location (name pending, see #74), the button text can be displayed beneath the icons:

![Single → Icon Labels With Shortcuts Menu → Desktop](https://user-images.githubusercontent.com/8181091/64555004-c0f08f00-d33c-11e9-9830-8169c68a88bb.jpg)

On desktop, when the "shortcuts" menu location is not set, we can move the button text up next to the icons and make the text a bit bigger (similar to the Chaplin structure):

![Single → Icon Labels Without Shortcuts Menu → Desktop](https://user-images.githubusercontent.com/8181091/64555162-0ca33880-d33d-11e9-88b3-54aeff041d41.jpg)

On mobile and tablet, when the shortcuts menu is not displayed, we can display the icon text below or next to the icons, depending on how much space space is available:

![Single → Icon Labels → Mobile](https://user-images.githubusercontent.com/8181091/64555225-280e4380-d33d-11e9-87ec-417706966e57.jpg)

The sizes of the elements might need to be adjusted a bit to allow sufficient space for the icon text in languages other than English languages.

**Comment by afercia** [comment]
@andersnoren thanks for addressing this issue, much appreciated (yup, I noticed on Chaplin there's visible text, haven't tried the "Shortcuts" menu though).
